src/main/java/com/wasilewskyy/medical_clinic/validation/PatientValidator.java

```diff
@@ -0,0 +1,40 @@
+package com.wasilewskyy.medical_clinic.validation;
+
+import com.wasilewskyy.medical_clinic.model.Patient;
+import com.wasilewskyy.medical_clinic.repository.PatientRepository;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public final class PatientValidator {
+
+    public static void validateNewPatient(Patient patient, PatientRepository patientRepository) {
+        if (patient.getIdCardNo() == null || patient.getFirstName() == null
+                || patient.getLastName() == null || patient.getBirthday() == null
+                || patient.getPhoneNumber() == null || patient.getEmail() == null
+                || patient.getPassword() == null) {
+            throw new IllegalArgumentException("None of the patient fields can be blank.");
+        }
+        if (patientRepository.findByEmail(patient.getEmail())) {
```

Comment:
brakuje sprawdzenia czy to nie jest przypadkiem ten sam pacjent -> jestem apcjentem o mailu 1234 nie zmieniam swojego maila wiec w body dostaje 1234